dataverse-persistence/src/main/resources/db/migration/V18__addedConsentEntities.sql

```diff
@@ -0,0 +1,35 @@
+create table consent
+(
+    id serial not null
+        constraint consent_pkey
+            primary key,
+    name varchar(255)
+        constraint consent_name_key
+            unique
+);
+
+create table consentaction
+(
+    id serial not null
+        constraint consentaction_pkey
+            primary key,
+    actionoptions json,
+    consentactiontype text not null,
+    consent_id bigint
+        constraint fk_consentaction_consent_id
+            references consent
+);
+
+create table consentdetails
+(
+    id serial not null
+        constraint consentdetails_pkey
+            primary key,
+    text text not null,
+    textlanguage varchar(255) not null,
+    hidden boolean not null,
+    required boolean not null,
```

Comment:
We could add `default` values for these two, and possibly for `locale` field aswell.

Reply:
Since boolean is by default false in java, it will be the default value in the moment of creating an entity. I wouldn't set locale to any default, I would prefer to see an error during save operation than deal with 3 consents with the same language afterwards.

Reply:
@MrK191 "having additional security restriction is not a negative" :P You will get an error message if something goes wrong on save and the transaction will be rolledback. As for `locale` in similar situation for `notificationslanguage` variable I was asked by @madryk to add default value for db consistency.

Reply:
I actually forgot that primary key is not null by default in most db's so that's what I meant :P. Well I guess if we omit JPA then there wouldn't be any default so I will add it.
About notification language, I think it is a different case since it is not really an critical function and the user can easily change it if he wants to. In this case if there was no error during save there would be an duplicate language in the same consent. And then there would be a big problem since the registration would break, because we wouldn't know which one to present tot he user. You would have to manually enter db and check the text to confirm the language.

Reply:
In the case of notification nothing breaks.

Reply:
Ok, but since we doesn't include `locale` in table primary key, we should take care in code about a possibility that we can have more than one `consentDetail` for a given `consent` with same `locale`. 
So this logic must exist anyway. I'll leave it up to you.

Reply:
I mean, you can have the same locale for the same consent right now.